tests/components/google_wifi/test_sensor.py

```diff
@@ -34,185 +31,181 @@
-            test_name = self.sensor_dict[name]["name"]
-            assert test_name == sensor.name
-
-    def test_unit_of_measurement(self):
-        """Test the unit of measurement."""
-        for name in self.sensor_dict:
-            sensor = self.sensor_dict[name]["sensor"]
-            assert self.sensor_dict[name]["units"] == sensor.unit_of_measurement
-
-    def test_icon(self):
-        """Test the icon."""
-        for name in self.sensor_dict:
-            sensor = self.sensor_dict[name]["sensor"]
-            assert self.sensor_dict[name]["icon"] == sensor.icon
-
-    @requests_mock.Mocker()
-    def test_state(self, mock_req):
-        """Test the initial state."""
-        self.setup_api(MOCK_DATA, mock_req)
-        now = datetime(1970, month=1, day=1)
-        with patch("homeassistant.util.dt.now", return_value=now):
-            for name in self.sensor_dict:
-                sensor = self.sensor_dict[name]["sensor"]
-                self.fake_delay(2)
-                sensor.update()
-                if name == google_wifi.ATTR_LAST_RESTART:
-                    assert "1969-12-31 00:00:00" == sensor.state
-                elif name == google_wifi.ATTR_UPTIME:
-                    assert 1 == sensor.state
-                elif name == google_wifi.ATTR_STATUS:
-                    assert "Online" == sensor.state
-                else:
-                    assert "initial" == sensor.state
-
-    @requests_mock.Mocker()
-    def test_update_when_value_is_none(self, mock_req):
-        """Test state gets updated to unknown when sensor returns no data."""
-        self.setup_api(None, mock_req)
-        for name in self.sensor_dict:
-            sensor = self.sensor_dict[name]["sensor"]
-            self.fake_delay(2)
+        },
+    )
+    assert_setup_component(6, "sensor")
+
+
+"""Tests for Google Wifi sensor platform."""
+
+
+def setup_api(data, requests_mock):
+    """Set up API with fake data."""
+    resource = f"http://localhost{google_wifi.ENDPOINT}"
+    now = datetime(1970, month=1, day=1)
+    with patch("homeassistant.util.dt.now", return_value=now):
+        requests_mock.get(resource, text=data, status_code=200)
+        conditions = google_wifi.MONITORED_CONDITIONS.keys()
+        API = google_wifi.GoogleWifiAPI("localhost", conditions)
+    NAME = "foo"
+    for condition, cond_list in google_wifi.MONITORED_CONDITIONS.items():
+        sensor = google_wifi.GoogleWifiSensor(API, NAME, condition)
+        name = f"{NAME}_{condition}"
+        units = cond_list[1]
+        icon = cond_list[2]
+        SENSOR_DICT[condition] = {
+            "sensor": sensor,
+            "name": name,
+            "units": units,
+            "icon": icon,
+        }
+    return API
+
+
+def fake_delay(hass, ha_delay):
+    """Fake delay to prevent update throttle."""
+    hass_now = dt_util.utcnow()
+    shifted_time = hass_now + timedelta(seconds=ha_delay)
+    hass.bus.fire(ha.EVENT_TIME_CHANGED, {ha.ATTR_NOW: shifted_time})
```

Comment:
We have a test helper for this:
https://github.com/home-assistant/core/blob/00847db54f00a821f044243696a09638a1847aa2/tests/common.py#L287-L288

Reply:
Bump.